k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -175,19 +155,17 @@ public void run() {
                     long startUid = getStartUid(oldUidNext, newUidNext);
 
                     if (newUidNext > startUid) {
-                        notifyMessagesArrived(startUid, newUidNext);
+                        pushReceiver.syncFolder(getName());
                     } else {
-                        processStoredUntaggedResponses();
-
                         if (K9MailLib.isDebug()) {
                             Timber.i("About to IDLE for %s", getLogId());
                         }
 
                         prepareForIdle();
 
-                        ImapConnection conn = connection;
+                        ImapConnection conn = folder.getConnection();
                         setReadTimeoutForIdle(conn);
```

Comment:
There are some places where it is appropriate to manage the connection directly, like `setReadTimeout` and not through the folder. I decided to create a new class `IdleConnectionManager` that wraps an `ImapConnection` object and exposes only the necessary method to `ImapFolderPusher`. I also moved all the code from the `IdleStopper` class into it.